src/gui/properties/peerlistwidget.cpp

```diff
@@ -138,6 +136,67 @@ PeerListWidget::~PeerListWidget()
     delete m_copyHotkey;
 }
 
+void PeerListWidget::displayToggleColumnsMenu(const QPoint&)
+{
+    QMenu hideshowColumn(this);
+    hideshowColumn.setTitle(tr("Column visibility"));
+    QList<QAction*> actions;
+    for (int i = 0; i < PeerListDelegate::IP_HIDDEN; ++i) {
```

Comment:
Inside the for loop do this:
```c++
if ((i == PeerListDelegate::COUNTRY) && !Preferences::resolvePeerCountries()) {
    actions.append(0); // null pointer to keep the index in sync. Didn't test if it will compile
    continue;
}
```